x-pack/plugins/security_solution/public/detections/components/rules/rule_preview/preview_histogram.tsx

```diff
@@ -67,45 +69,48 @@ export const PreviewHistogram = ({
   previewId,
   addNoiseWarning,
   spaceId,
-  threshold,
   ruleType,
   index,
 }: PreviewHistogramProps) => {
+  const dispatch = useDispatch();
   const { setQuery, isInitializing } = useGlobalTime();
   const { timelines: timelinesUi, cases } = useKibana().services;
   const from = useMemo(() => `now-1${timeFrame}`, [timeFrame]);
   const to = useMemo(() => 'now', []);
   const startDate = useMemo(() => formatDate(from), [from]);
   const endDate = useMemo(() => formatDate(to), [to]);
   const isEqlRule = useMemo(() => ruleType === 'eql', [ruleType]);
-  const isThresholdRule = useMemo(() => ruleType === 'threshold', [ruleType]);
+  const isMlRule = useMemo(() => ruleType === 'machine_learning', [ruleType]);
 
-  const [isLoading, { data, inspect, totalCount, refetch, buckets }] = usePreviewHistogram({
+  const [isLoading, { data, inspect, totalCount, refetch }] = usePreviewHistogram({
     previewId,
     startDate,
     endDate,
     spaceId,
-    threshold: isThresholdRule ? threshold : undefined,
     index,
     ruleType,
   });
 
   const {
-    columns,
-    dataProviders,
-    deletedEventIds,
-    kqlMode,
-    itemsPerPage,
-    itemsPerPageOptions,
-    graphEventId,
-    sort,
-  } = alertsDefaultModel;
+    timeline: {
+      columns,
+      dataProviders,
+      defaultColumns,
+      deletedEventIds,
+      graphEventId, // If truthy, the graph viewer (Resolver) is showing
```

Comment:
probably don't need this, and if you do, https://github.com/elastic/kibana/blob/main/x-pack/plugins/timelines/public/components/t_grid/integrated/index.tsx#L116 should be graphEventId?: string; instead